docs/dev/zksync-v2/contracts.md

```diff
@@ -1,25 +1,25 @@
 # Contract deployment
 
-To maintain the same security as L1, the zkSync operator must publish the contract code for each contract it deploys. However, if there are multiple contracts deployed with the same code, it needs to be published onchain only once.
+To maintain the same security as L1, the zkSync operator must publish on chain the contract code for each contract it deploys. However, if there are multiple contracts deployed with the same code, it needs to be published only once.
 
 While deploying contracts for the first time may be relatively expensive, factories, which deploy contracts with the same code multiple times, can have huge savings compared to L1.
 
-All these specifics make the process of deploying smart contracts on zkEVM comply with the major rule: _The operator should know the code of the contract before it is deployed_. That means that deploying contracts is only possible by the means of `EIP712` transactions with the `factory_deps` field containing the supplied bytecode. More on EIP712 transactions [here](../../api/api.md#eip712).
+All these specifics make the process of deploying smart contracts on zkEVM comply with the major rule: _The operator should know the code of the contract before it is deployed_. This means that deploying contracts is only possible by the means of `EIP712` transactions with the `factory_deps` field contains the supplied bytecode. More on EIP712 transactions [here](../../api/api.md#eip712).
```

Comment:
I don't have a strong opinion on `This/That`, but I disagree about `contains` and Grammarly disagrees as well :)

Reply:
I don't know why but github does not offer me to a "commit" button. I'll change this later

Reply:
Now I understand that this wasn't a suggestion...